### PR TITLE
Linting changes

### PR DESCRIPTION
**Pull request by gregschohn**
### Description
First wave of cleaning up some linting errors/warnings.
Along the way, I've made some slight improvements.
I've continued renaming the just moved testUtilities classes to testFixture.
StreamChannelConnectionCaptureSerializer no longer bothers to implement close() and AutoCloseable. Nobody was using it that way anyway. In its place are flushes within the netty request and response handlers when the handlers are removed from the pipeline.

* Category: Lint fixes and Refactoring
* What is the old behavior before changes and new behavior after changes? One minor user-visible difference...  Streams should be flushed when the CaptureProxy is shut down.

### Issues Resolved
None

### Testing
allTests and :dockerSolution:composeUp (to move the OSB 4 test workload), which failed after geonames.

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TrafficCapture/captureOffloader/src/main/java/org/opensearch/migrations/trafficcapture/StreamChannelConnectionCaptureSerializer.java [hunk 1]
```diff
@@ -190,40 +190,19 @@ public CompletableFuture<Object> flushCommitAndResetStream(boolean isFinal) thro
         return future;
     }
 
-    /**
-     * This call is BLOCKING.  Override the Closeable interface - not addCloseEvent.
-     *
-     * @throws IOException
-     */
-    @Override
-    public void close() throws IOException {
```
Comment (lewijacn) [inline TrafficCapture/captureOffloader/src/main/java/org/opensearch/migrations/trafficcapture/StreamChannelConnectionCaptureSerializer.java:199]: Surprised we don't call this anywhere. I guess we close the channel in places but never the Serializer...

File: TrafficCapture/nettyWireLogging/src/main/java/org/opensearch/migrations/trafficcapture/netty/LoggingHttpRequestHandler.java [hunk 2]
```diff
@@ -114,6 +114,21 @@ public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
         });
     }
 
+    @Override
+    public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
```
Comment (lewijacn) [inline TrafficCapture/nettyWireLogging/src/main/java/org/opensearch/migrations/trafficcapture/netty/LoggingHttpRequestHandler.java:118]: Why do we need this function now?

File: TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/netty/NettyScanningHttpProxy.java [hunk 3]
```diff
@@ -58,4 +58,8 @@ public void stop() throws InterruptedException {
             bossGroup.shutdownGracefully();
         }
     }
+
+    public void waitForClose() throws InterruptedException {
```
Comment (lewijacn) [inline TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/netty/NettyScanningHttpProxy.java:62]: This looks cleaner, and seems like it should work but makes me a little skeptical

File: TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/netty/ProxyChannelInitializer.java [hunk 4]
```diff
@@ -30,7 +30,8 @@ public boolean shouldGuaranteeMessageOffloading(HttpRequest httpRequest) {
         return (httpRequest != null &&
                 (httpRequest.method().equals(HttpMethod.POST) ||
                         httpRequest.method().equals(HttpMethod.PUT) ||
-                        httpRequest.method().equals(HttpMethod.DELETE)));
+                        httpRequest.method().equals(HttpMethod.DELETE) ||
+                        httpRequest.method().equals(HttpMethod.PATCH)));
```
Comment (lewijacn) [inline TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/netty/ProxyChannelInitializer.java:34]: Good catch

**Review by lewijacn — APPROVED** [review]
Approved with some minor comments

File: TrafficCapture/nettyWireLogging/src/main/java/org/opensearch/migrations/trafficcapture/netty/LoggingHttpRequestHandler.java (hunk 2 above)
Comment (gregschohn) [reply TrafficCapture/nettyWireLogging/src/main/java/org/opensearch/migrations/trafficcapture/netty/LoggingHttpRequestHandler.java:118]: To close the serializer - see the previous comment that you left :)

File: TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/netty/NettyScanningHttpProxy.java (hunk 3 above)
Comment (gregschohn) [reply TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/netty/NettyScanningHttpProxy.java:62]: worst case scenario is that it doesn't and the process is killed.  That's pretty much where we were before too w/ the inescapable loop.

**Review by gregschohn — COMMENTED** [review]
responding to comments

**Pull request merged**